modules/programs/rofi/default.nix

```diff
@@ -0,0 +1,90 @@
+{ mode, config, pkgs, lib, ... }:
+
+with lib;
+let
+  i3Package =
+    if mode == "home-manager" then config.xsession.windowManager.i3.package
+    else config.services.xserver.windowManager.i3.package;
+
+  i3Support = pkgs.stdenvNoCC.mkDerivation rec {
+    pname = "rofi-i3-support";
+    version = "0.0.1";
+
+    src = ./i3-support;
+
+    installPhase = ''
+      install -d -m755 $out/bin $out/lib
+
+      substitute $src/i3-move-container.sh $out/bin/i3-move-container \
+        --subst-var-by i3-msg_bin ${i3Package}/bin/i3-msg \
+        --subst-var-by jq_bin ${pkgs.jq}/bin/jq \
+        --subst-var-by out_dir $out
+
+      substitute $src/i3-rename-workspace.sh $out/bin/i3-rename-workspace \
+        --subst-var-by i3-msg_bin ${i3Package}/bin/i3-msg \
+        --subst-var-by jq_bin ${pkgs.jq}/bin/jq \
+        --subst-var-by out_dir $out
+
+      substitute $src/i3-switch-workspaces.sh $out/bin/i3-switch-workspaces \
+        --subst-var-by i3-msg_bin ${i3Package}/bin/i3-msg \
+        --subst-var-by jq_bin ${pkgs.jq}/bin/jq \
+        --subst-var-by out_dir $out
+
+      substitute $src/list-workspaces.sh $out/lib/list-workspaces.sh \
+        --subst-var-by i3-msg_bin ${i3Package}/bin/i3-msg \
+        --subst-var-by jq_bin ${pkgs.jq}/bin/jq \
+        --subst-var-by out_dir $out
+
+      chmod 755 $out/bin/*
+    '';
+  };
+
+  cfg = config.soxin.programs.rofi;
+in
+{
+  options = {
+    soxin.programs.rofi = {
+      enable = mkEnableOption "rofi";
+
+      i3Support = mkOption {
+        type = types.bool;
+        default = false;
+        example = true;
+        description = ''
+          Whether to enable i3 support for rofi.
+
+          When enabled, you can set i3 bindings to the following commands:
+          exec ''${pkgs.rofi}/bin/rofi -show i3Workspaces
+          exec ''${pkgs.rofi}/bin/rofi -show i3RenameWorkspace
+          exec ''${pkgs.rofi}/bin/rofi -show i3MoveContainer
+        '';
+      };
+
+      dpi = mkOption {
+        type = with types; nullOr ints.positive;
+        default = null;
+        description = "The DPI of the rofi.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    (optionalAttrs (mode == "home-manager") {
+
+      programs.rofi = {
+        enable = true;
+
+        extraConfig = ''
+          ${optionalString cfg.i3Support ''
+          rofi.modi: window,run,ssh,drun,i3Workspaces:${i3Support}/bin/i3-switch-workspaces,i3RenameWorkspace:${i3Support}/bin/i3-rename-workspace,i3MoveContainer:${i3Support}/bin/i3-move-container
```

Comment:
I don't think this is appropriate as not everyone would want these modules supported. Why not use the concept of the plugin and let them configure it as they see fit?

https://github.com/nix-community/home-manager/blob/bcbf09a2021712a501681fa984f659fddad3c2e8/modules/programs/rofi.nix#L141